### PR TITLE
fix(#4850): Deno.remove() fails to remove unix socket

### LINKED ISSUE #4850: Deno.remove() fails to remove unix socket
Deno 0.41, Mac 10.14.6

If you have a unix domain socket `sock`, `Deno.remove("sock")` results in 

```
Error: Not a directory (os error 20)
``` 

I tried adding the `{recusrsive:true}` flag, but that changed nothing. I am able to remove regular text files without problem.

Edit: the socket is not in use, and I'm able to remove with `rm sock` naturally.

### PR DESCRIPTION
**Pull request by uki00a**
This PR closes #4850.

### CODE REVIEW

File: cli/tests/unit/remove_test.ts [hunk 1]
```diff
@@ -460,6 +483,29 @@ unitTest({ perms: { write: false } }, async function removeAllPerm(): Promise<
...
+    await Deno.remove(path);
+    let err;
+    try {
+      Deno.statSync(path);
+    } catch (e) {
+      err = e;
+    }
+    assert(err instanceof Deno.errors.NotFound);
+  }
+);
```
Comment (ry) [inline cli/tests/unit/remove_test.ts:507]: Why are there two copies of the tests?

Comment (uki00a) [reply cli/tests/unit/remove_test.ts:507]: I'm sorry, it's a bit confusing.
One is a test of `Deno.removeSync` and the other is a test of `Deno.remove`.

**Comment by uki00a** [comment]
@ry I removed duplicate codes between sync and async cases. Please review it again.
(Actually, there are other similar duplicates in that test file, but I haven't fixed them in this PR because it would complicate things.)

File: cli/tests/unit/remove_test.ts (hunk 1 above)
Comment (ry) [reply cli/tests/unit/remove_test.ts:486]: Ah sorry, I didn't catch that. This is fine too. Thanks.

**Comment by uki00a** [comment]
@ry Thanks!

**Pull request merged**